statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,257 @@
+import numpy as np 
+import pandas as pd
+#from numbers import Number
+
+
+class SurveyStat(object):
+
+    def __init__(self, data, cluster=None, strata=None, prob_weights=None):
+
+        data, self.cluster = self._check_type(data, cluster)
+        data, self.strata = self._check_type(data, strata)
+        data, self.prob_weights = self._check_type(data, prob_weights)
+        
+        self.p = data.shape[1]
+        self.data = np.asarray(data)
+
+    # preprocessing to be used by init
+    def _check_type(self, data, vname):
+        """
+        converts cluster, strata, etc to an numpy array
+        gets rid of column from data if possible
+
+        Parameters
+        ----------
+        data : array-like
+          The raw data
+        vname : one of cluster, strata, or prob_weight
```

Comment:
Since vname can be either the name of the variable (a string) or the data itself, I think this variable name should be changed.  Some people like names like `maybe_name_or_data` but I prefer something shorter, like `var`.